hack/generated/controllers/crd_resourcegroup_test.go

```diff
@@ -12,17 +12,22 @@ import (
 	. "github.com/onsi/gomega"
 
 	"github.com/Azure/k8s-infra/hack/generated/pkg/armclient"
+	util "github.com/Azure/k8s-infra/hack/generated/pkg/util"
 )
 
 func Test_ResourceGroup_CRUD(t *testing.T) {
 	t.Parallel()
 
 	g := NewGomegaWithT(t)
+	log := util.NewBannerLogger()
+	log.Header(t.Name())
```

Comment:
Discussion here: https://github.com/golang/go/issues/19280 suggests a few things to me:

Firstly, `== CONT <TESTNAME>` is always logged as the first line of a test already. It's not clear to me that another line directly below that is all that helpful (even if it is a bit easier to read it also requires test authors to manually write that long line in their tests which they're likely to forget).

Secondly (from the linked issue above): 
> In order to see logging interleaved in a more structured way with test output, one would use t.Log, not println, log, or fmt.
> ...
> If you use t.Log, the logs will be grouped with the respective tests. In this case, tests collect the logs for this test and flush it to the parent on completion. Top-level tests flush to stdout.

The `BannerLogger` is not using `t.Log` which means that it will be outputting logs that may end up interleaved or separated from the tests in question (I think? @babbageclunk maybe knows more about this?).

Additionally, I am not sure that the log output from the tests themselves (which is really coming from the controllers hosted by envtest) is going to be located in the right place, because it _also_ isn't using `t.Log` as far as I can tell.

The `controllers.RegisterAll` function already takes a `log logr.Logger` parameter, and I am wondering if all we really need to do to make the test output more clear is to supply one of those that ends up routing to `t.Log`. That should remove any interleaving going on in test logs as well. We call `controllers.RegisterAll` in `kube_test_context_envtest.go`.

Reply:
Agree that we should be using the facilities provided by `testing.T` and not `fmt`.